File: src/main/java/com/pomankao/Hello.java
package com.pomankao;

//import com.pomankao.student.Student;

import com.kotlin.Student;

import java.util.Scanner;

public class Hello {
    public static void main(String[] args) {
//        System.out.println("Hello world");
//        String s = "abcde";
        // test Student.java default access modifier
//        Student student = new Student("Poman",50,30);
//        student.testDefaultAccessModifier(); // Can't be access
//        testDefaultAccessModifier(); // Can't be access
        // test Student.kt
        Student student = new Student("Poman",50,30);
        // test internal access modifier
//        student.grading(); // Can't be access because they are different module.


        String s = null;
        System.out.println(s.charAt(3));
        System.out.println(s.length());
       Person person = new Person(66.5f, 1.7f);
        person.hello();

        int score = 88;
        System.out.println(score < 80 || score > 90); // false
        char c = 'A'; // ASCII: 65
        System.out.println(c > 'a'); // false
        System.out.print("Please enter number: ");
        Scanner scanner = new Scanner(System.in);
        int number = scanner.nextInt();
        System.out.println(number > 60);
    }
}

File: src/main/kotlin/com/pomankao/Hello.kt
package com.pomankao

import com.kotlin.Student

fun main(args: Array<String>) {
//    println("Hello Kotlin")
    val human = Human("Poman",80.5f,1.8f)
    human.hello()

//    var s = "abcde"
//    s = null  // Null can't be a value
    var s:String? = "abcde"
    s = null
    println(s!!.length)
    println(s?.get(3))
    println(s?.substring(3))

    // test internal access modifier
    val stu = Student("Poman",50,30)
    stu.grading() // Can access
}


class Human(var name: String, var weight: Float, var height: Float) {

    fun bmi(): Float {
        return weight/(height*height)
    }

    fun hello() {
        println("Hello kotlin")
    }
}